pkg/controller/projects/projects/project.go

```diff
@@ -149,6 +158,52 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	if !ok {
 		return managed.ExternalCreation{}, errors.New(errNotProject)
 	}
+	// Line 161 -> 206 will be export to an util function
+	keySecretRef := cr.Spec.ForProvider.ImportUrlSecretRef
+	importUrl := cr.Spec.ForProvider.ImportURL
+
+	// User has provided secret for the import
+	if keySecretRef != nil && importUrl != nil {
+		// Retrieve secret from k8s
+		namespacedName := types.NamespacedName{
+			Namespace: keySecretRef.Namespace,
+			Name:      keySecretRef.Name,
+		}
+
+		secret := &corev1.Secret{}
+
+		err := e.kube.Get(ctx, namespacedName, secret)
+
+		if err != nil {
+			return managed.ExternalCreation{}, errors.New(errNoImportSecret)
+		}
+
+		// Obtain the password from the secret.
+		password := string(secret.Data[keySecretRef.PasswordKey])
+		if password == "" {
+			return managed.ExternalCreation{}, errors.New(errPasswordNotFound)
+		}
+
+		// Obtain the username from the secret.
+		username := string(secret.Data[keySecretRef.UsernameKey])
+		if username == "" {
+			return managed.ExternalCreation{}, errors.New(errUsernameNotFound)
+		}
+
+		// manipulate url to add the secret. If secret is already in the url, it should be overridden
+		// http://username:password@example.com
+		parsedUrl, err := url.Parse(*importUrl)
+
+		if err != nil {
+			return managed.ExternalCreation{}, errors.New(errUsernameNotFound)
```

Comment:
Wrong error. If we were not able to parse the initially provided URL, the URL is probably in an incorrect format, as we can not parse it here.
Error with a message, displaying an issue with parsing of the provided URL. Also wrap the provided `err` to provide an actual trace of failure context.